{
 "cells": [
  {
<<<<<<< HEAD
   "cell_type": "markdown",
   "id": "fe762568-8a91-451e-adf6-698586a15966",
   "metadata": {},
   "source": [
    "# Finding New Proteins"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
=======
   "cell_type": "code",
   "execution_count": 121,
>>>>>>> 85b12dc514aed880607e74140170a6fe33e6f95d
   "id": "b28ad0b5-0f33-4267-a43f-801d8d278f61",
   "metadata": {},
   "outputs": [],
   "source": [
    "import csv\n",
    "from datetime import datetime\n",
    "\n",
<<<<<<< HEAD
    "with open(\"Source/ALLbrakerAA against human.txt\") as f:\n",
    "    reader1 = csv.reader(f, delimiter=\"\\t\")\n",
    "    new = list(reader1)\n",
    "\n",
    "with open(\"Source/v10.7 against human.txt\") as e:\n",
    "    reader2 = csv.reader(e, delimiter=\"\\t\")\n",
    "    ori = list(reader2)\n",
    "\n",
    "with open(\"Source/brakerAA against v10.7.txt\") as x:\n",
    "    reader3 = csv.reader(x, delimiter=\"\\t\")\n",
    "    xen = list(reader3)\n",
    "\n",
    "output = \"Results/output.txt\" #this is the raw list of NOVEL proteins that have not been mapped to before by v10.7\n",
    "matchedproteins = \"Results/matchedProteins.txt\" #this is a SUBSET of output.txt: braker transcripts which mapped to v10.7 transcripts AND have hits for the corresponding v10.7 query\n",
    "newproteins = \"Results/newProteins.txt\" #this is a SUBSET of output.txt: braker transcripts which mapped to v10.7 transcripts BUT DON'T have hits for the corresponding v10.7 query\n",
    "npnt = \"Results/newProteinsNewTranscript.txt\" #this is a SUBSET of output.txt: braker transcripts which DID NOT map to v10.7 transcripts\n",
    "npntpro = \"Results/newProteinsNewTranscript_ListofProteins.txt\" #this is the unique list of proteins in newProteinsNewTranscript.txt"
=======
    "with open(\"ALLbrakerAA against human.txt\") as f:\n",
    "    reader1 = csv.reader(f, delimiter=\"\\t\")\n",
    "    new = list(reader1)\n",
    "\n",
    "with open(\"v10.7 against human.txt\") as e:\n",
    "    reader2 = csv.reader(e, delimiter=\"\\t\")\n",
    "    ori = list(reader2)\n",
    "\n",
    "with open(\"brakerAA against v10.7.txt\") as x:\n",
    "    reader3 = csv.reader(x, delimiter=\"\\t\")\n",
    "    xen = list(reader3)\n",
    "\n",
    "output = \"output.txt\" #this is the raw list of NOVEL proteins that have not been mapped to before by v10.7\n",
    "matchedproteins = \"matchedProteins.txt\" #this is a SUBSET of output.txt: braker transcripts which mapped to v10.7 transcripts AND have hits for the corresponding v10.7 query\n",
    "newproteins = \"newProteins.txt\" #this is a SUBSET of output.txt: braker transcripts which mapped to v10.7 transcripts BUT DON'T have hits for the corresponding v10.7 query\n",
    "npnt = \"newProteinsNewTranscript.txt\" #this is a SUBSET of output.txt: braker transcripts which DID NOT map to v10.7 transcripts\n",
    "npntpro = \"newProteinsNewTranscript_ListofProteins.txt\" #this is the unique list of proteins in newProteinsNewTranscript.txt"
>>>>>>> 85b12dc514aed880607e74140170a6fe33e6f95d
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 122,
   "id": "631350db-ad1c-4b8d-aabb-403a3fa5dafd",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "P2RX7\n",
      "5.1e-131\n"
     ]
    }
   ],
   "source": [
    "print (new[0][2])\n",
    "print (ori[0][4])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 123,
   "id": "9193ddba-397c-4821-a155-d64c3a2cb719",
   "metadata": {},
   "outputs": [],
   "source": [
    "# compare human HMMER results for braker vs. v10.7\n",
    "# essentially, scans through the list of proteins mapped to by braker and \n",
    "# returns a truncated list of NOVEL PROTEINS that have NOT been mapped to by v10.7\n",
    "max = len(ori)\n",
    "maxnew = len(new)\n",
    "i = 0\n",
    "j = 0\n",
    "res = []\n",
    "\n",
    "while (j<maxnew):\n",
    "    while (i<max):\n",
    "        if new[j][2] == ori[i][2]:\n",
    "            i = max\n",
    "            #print(\"match found for \" + new[j][0])\n",
    "        i+=1\n",
    "        if i == max:\n",
    "            res.append(new[j])\n",
    "    j+=1\n",
    "    i=0"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 124,
   "id": "758da54c-b348-4985-b247-c9abf053b842",
   "metadata": {},
   "outputs": [],
   "source": [
    "# this is the raw list of braker transcripts that HMMER'ed to novel human proteins NOT mapped to before by v10.7\n",
    "with open(output, \"w\") as out:\n",
    "    now = datetime.now()\n",
    "    out.write(now.strftime(\"%m/%d/%Y, %H:%M:%S\") + '\\n')\n",
    "\n",
    "with open(output, \"a\") as out:\n",
    "    for newgene in res:\n",
    "        for item in newgene:\n",
    "            out.write(item + '\\t')\n",
    "        out.write('\\n')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 125,
   "id": "0a97b19f-05b8-4af7-ab3b-63365fd3d01d",
   "metadata": {},
   "outputs": [],
   "source": [
    "with open(\"output.txt\") as out1:\n",
    "    reader4 = csv.reader(out1, delimiter=\"\\t\")\n",
    "    outr = list(reader4)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 126,
   "id": "118f19aa-9d91-45e6-9fd3-284d6bbe73e2",
   "metadata": {},
   "outputs": [],
   "source": [
    "# of the \"new proteins\", compare braker HMMER with v10.7 HMMER if the braker transcripts matches to a v10.7 gene\n",
    "# if the braker transcript has no mapping to a v10.7 gene, then compile to newProteinsNewTranscript.txt\n",
    "maxxen = len(xen)\n",
    "maxout = len(outr)\n",
    "i = 0\n",
    "j = 0\n",
    "res2 = []\n",
    "res3 = []\n",
    "res4 = []\n",
    "\n",
    "while (j<maxout):\n",
    "    while (i<maxxen):\n",
    "        if outr[j][0] == xen[i][0]:\n",
    "            #search for v10.7 HMMR results against human\n",
    "            orit = list(map(list, zip(*ori)))\n",
    "            if xen[i][1] in orit[0]:\n",
    "                key = orit[0].index(xen[i][1])\n",
    "                temparray = [outr[j][0], outr[j][2], outr[j][4], ori[key][0], ori[key][2], ori[key][4]]\n",
    "                res2.append(temparray)\n",
    "                i = maxxen\n",
    "            else:\n",
    "                temparray = [outr[j][0], outr[j][2], outr[j][4], xen[i][1]]\n",
    "                res3.append(temparray)\n",
    "                i = maxxen\n",
    "        i+=1\n",
    "        if i == maxxen and j != 0:\n",
    "            res4.append(outr[j])\n",
    "    j+=1\n",
    "    i=0"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 127,
   "id": "504b75a2-fe42-446e-9c3a-057e0d1cfb52",
   "metadata": {},
   "outputs": [],
   "source": [
    "# these transcripts have sequence similarity to v10.7\n",
    "# the corresponding v10.7 sequence DOES NOT HAVE a significant mapping to a human protein\n",
    "# the output will include BrakerID | HMMERed Gene(Braker) | e-value | v10.7ID\n",
    "with open(newproteins, \"w\") as out:\n",
    "    now = datetime.now()\n",
    "    out.write(now.strftime(\"%m/%d/%Y, %H:%M:%S\") + '\\n')\n",
    "\n",
    "with open(newproteins, \"a\") as out:\n",
    "    for newgene in res3:\n",
    "        for item in newgene:\n",
    "            out.write(item + '\\t')\n",
    "        out.write('\\n')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 128,
   "id": "2eda171b-1b3f-461b-932e-433f426176d2",
   "metadata": {},
   "outputs": [],
   "source": [
    "# these transcripts have sequence similarity to v10.7\n",
    "# the corresponding v10.7 sequence already has a significant mapping to a human protein\n",
    "# the output will include BrakerID | HMMERed Gene(Braker) | e-value | v10.7ID | HMMERed Gene(v10.7) | e-value\n",
    "with open(matchedproteins, \"w\") as out:\n",
    "    now = datetime.now()\n",
    "    out.write(now.strftime(\"%m/%d/%Y, %H:%M:%S\") + '\\n')\n",
    "\n",
    "with open(matchedproteins, \"a\") as out:\n",
    "    for newgene in res2:\n",
    "        for item in newgene:\n",
    "            out.write(item + '\\t')\n",
    "        out.write('\\n')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 129,
   "id": "325ae68f-5177-4198-8372-543d895f7998",
   "metadata": {},
   "outputs": [],
   "source": [
    "# these transcripts have NO sequence similarity to v10.7\n",
    "with open(npnt, \"w\") as out:\n",
    "    now = datetime.now()\n",
    "    out.write(now.strftime(\"%m/%d/%Y, %H:%M:%S\") + '\\n')\n",
    "\n",
    "with open(npnt, \"a\") as out:\n",
    "    for newgene in res4:\n",
    "        for item in newgene:\n",
    "            out.write(item + '\\t')\n",
    "        out.write('\\n')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 130,
   "id": "81c3e756-2486-4103-afe5-8d6ed2e9b6fd",
   "metadata": {},
   "outputs": [],
   "source": [
    "# finally, get list of unique proteins from the newProteinNewTranscript file\n",
    "with open(\"newProteinsNewTranscript.txt\") as out:\n",
    "    reader5 = csv.reader(out, delimiter=\"\\t\")\n",
    "    npntr = list(reader5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 131,
   "id": "45a13282-9752-4387-b5d3-16b42b90ba9b",
   "metadata": {},
   "outputs": [],
   "source": [
    "npntr.remove(npntr[0])\n",
    "npntt = list(map(list, zip(*npntr)))\n",
    "npntgenes = npntt[2]\n",
    "npntgenes = list(dict.fromkeys(npntgenes))\n",
    "npntgenes.sort()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 132,
   "id": "f432810d-d35a-4f31-ae53-7642e18aa89a",
   "metadata": {},
   "outputs": [],
   "source": [
    "# these are the new proteins from braker transcripts that did not map to any v10.7 genes significantly\n",
    "with open(npntpro, \"w\") as out:\n",
    "    now = datetime.now()\n",
    "    out.write(now.strftime(\"%m/%d/%Y, %H:%M:%S\") + '\\n')\n",
    "\n",
    "with open(npntpro, \"a\") as out:\n",
    "    for newgene in npntgenes:\n",
    "        out.write(newgene + '\\n')"
   ]
  },
  {
<<<<<<< HEAD
   "cell_type": "markdown",
   "id": "6e845276-5116-4c53-a06b-aa24642ea0c3",
   "metadata": {},
   "source": [
    "# Verify Selenoproteins"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "4154f6ff-0696-4b3c-9cb9-775213c48d69",
   "metadata": {},
   "outputs": [],
   "source": [
    "import csv\n",
    "from datetime import datetime"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "abdf88da-9acb-4dfd-b754-b7a2573a45ab",
   "metadata": {},
   "outputs": [],
   "source": [
    "with open(\"Source/selenoproteins10.7.txt\") as x:\n",
    "    reader3 = csv.reader(x, delimiter=\"\\t\")\n",
    "    seleno = list(reader3)\n",
    "\n",
    "with open(\"Source/brakerAA against v10.7.txt\") as x:\n",
    "    reader3 = csv.reader(x, delimiter=\"\\t\")\n",
    "    xen = list(reader3)\n",
    "\n",
    "with open(\"Source/braker.aa\") as x:\n",
    "    reader3 = csv.reader(x, delimiter=\"\\t\")\n",
    "    braker = list(reader3)\n",
    "\n",
    "with open(\"Source/ALLbrakerAA against human.txt\") as f:\n",
    "    reader1 = csv.reader(f, delimiter=\"\\t\")\n",
    "    new = list(reader1)\n",
    "\n",
    "with open(\"Source/v10.7 against human.txt\") as f:\n",
    "    reader1 = csv.reader(f, delimiter=\"\\t\")\n",
    "    tenpseven = list(reader1)\n",
    "\n",
    "with open(\"Source/Human Proteins with Selenocysteines.fasta\") as f:\n",
    "    reader1 = csv.reader(f, delimiter=\"\\t\")\n",
    "    humanseleno = list(reader1)\n",
    "\n",
    "with open(\"Source/Xentr10.7.allTrs.pep.fasta.txt\") as f:\n",
    "    reader1 = csv.reader(f)\n",
    "    xentraa = list(reader1)\n",
    "\n",
    "i = 0\n",
    "j = 0\n",
    "k = 0\n",
    "max = len(xen)\n",
    "selenon = len(seleno)\n",
    "maxbraker = len(braker)\n",
    "maxnew = len(new)\n",
    "maxtenpseven = len(tenpseven)\n",
    "maxhumanseleno = len(humanseleno)\n",
    "maxxentraa = len(xentraa)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "1edb066e-6b3a-427f-9ad7-cedf17f84391",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "PERFECT MATCHES: \n",
      ">Xetrov107022849m\n",
      ">Xetrov107025172m\n",
      ">Xetrov107031781m\n",
      ">Xetrov107038964m\n",
      "\n",
      "OKAY MATCHES:\n",
      ">Xetrov107041767m\n",
      "\n",
      "BREAKER TRUNCATED:\n",
      ">Xetrov107005635m\n",
      ">Xetrov107051203m\n",
      ">Xetrov107034106m\n",
      "\n",
      "NO MATCHES:\n",
      ">Xetrov107008048m\n",
      ">Xetrov107032545m\n",
      ">Xetrov107034602m (quite short...)\n",
      ">Xetrov107041207m\n"
     ]
    }
   ],
   "source": [
    "# THIS METHOD ASSUMES THAT 10.7 IS CORRECT. I HAVE SUSPICIONS ABOUT THAT!\n",
    "matched = []\n",
    "i = 0\n",
    "j = 0\n",
    "k = 0\n",
    "while (j<max):\n",
    "    while (i<selenon):\n",
    "        selenogene = seleno[i][0][1:]\n",
    "        if xen[j][1] == selenogene:\n",
    "            k = 0\n",
    "            #print(\"\\n\" + seleno[i][0] + \" sequence from 10.7:\\n\" + seleno[i+1][0])\n",
    "            numu = seleno[i+1][0].count(\"C\")\n",
    "            #print(\"C appeared: \" + str(numu) + \" times\")\n",
    "            while (k<maxbraker):\n",
    "                if k%3!=2:\n",
    "                    if xen[j][0] == braker[k][0][1:]:\n",
    "                        #print(\">\" + xen[j][0] + \" Sequence from Breaker:\\n\" + braker[k+1][0])\n",
    "                        numu = braker[k+1][0].count(\"C\")\n",
    "                        #print(\"C appeared: \" + str(numu) + \" times\")\n",
    "                        if seleno[i+1][0] == braker[k+1][0]:\n",
    "                            #print(\"PERFECT MATCH FOR \" + seleno[i][0] + \": \" + \">\" + xen[j][0])\n",
    "                            matched.append(seleno[i][0])\n",
    "                k+=1\n",
    "            #print(\"match found for \" + new[j][0])\n",
    "        i+=2\n",
    "    j+=1\n",
    "    i=0\n",
    "print(\"\\nPERFECT MATCHES: \")\n",
    "for n in matched:\n",
    "    print(n)\n",
    "\n",
    "print(\"\\nOKAY MATCHES:\\n>Xetrov107041767m\")\n",
    "print(\"\\nBREAKER TRUNCATED:\\n>Xetrov107005635m\\n>Xetrov107051203m\\n>Xetrov107034106m\")\n",
    "print(\"\\nNO MATCHES:\\n>Xetrov107008048m\\n>Xetrov107032545m\\n>Xetrov107034602m (quite short...)\\n>Xetrov107041207m\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "1b77ae44-db12-4c94-9186-02705ae6f062",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['g10478.t1', 'sp|Q9BVL4|SELO_HUMAN', 'SELENOO', 'Selenoprotein O ', '2.2e-100', '']\n",
      "['g10478.t2', 'sp|Q9BVL4|SELO_HUMAN', 'SELENOO', 'Selenoprotein O ', '2.7e-100', '']\n",
      "['g1679.t1', 'sp|Q8WWX9|SELM_HUMAN', 'SELENOM', 'Selenoprotein M ', '2.9e-23', '1.2e-23']\n",
      "['g2069.t1', 'sp|Q8WWX9|SELM_HUMAN', 'SELENOM', 'Selenoprotein M ', '9e-11', '']\n",
      "['g5490.t1', 'sp|Q9BVL4|SELO_HUMAN', 'SELENOO', 'Selenoprotein O ', '8.7e-284', '']\n",
      "['g5490.t2', 'sp|Q9BVL4|SELO_HUMAN', 'SELENOO', 'Selenoprotein O ', '8.7e-284', '8e-283']\n",
      "['g6082.t1', 'sp|Q9BQE4|SELS_HUMAN', 'SELENOS', 'Selenoprotein S ', '2.4e-57', '1.6e-57']\n",
      "['g7697.t1', 'sp|P49908|SEPP1_HUMAN', 'SELENOP', 'Selenoprotein P ', '2.4e-55', '7.9e-54']\n",
      "['g9782.t1', 'sp|P62341|SELT_HUMAN', 'SELENOT', 'Thioredoxin reductase-like selenoprotein T ', '7.8e-88', '3.8e-88']\n",
      "g10478.t1 preserves the USS as CSS\n",
      "g10478.t2 preserves the USS as CSS\n",
      "g1679.t1 sequence is AFTER the selenocysteine is (see alignment)\n",
      "g2069.t1 is not very similar and the U is skipped but the rest of the protein is translated (see alignment)\n",
      "g5490.t1 cut off exactly where U is (mistranslation of selenocysteine likely)\n",
      "g5490.t2 cut off exactly where U is (mistranslation of selenocysteine likely)\n",
      "g6082.t1 cut off exactly where U is (mistranslation of selenocysteine likely)\n",
      "g7697.t1 preserves the first ASUYL as ASCGF, but the 9 selenocysteines in the C-terminus are truncated (see alignement)\n",
      "g9782.t1 sequence is AFTER the selenocysteine is (see alignment)\n"
     ]
    }
   ],
   "source": [
    "# for Breaker sequences\n",
    "i = 0\n",
    "while i < maxnew:\n",
    "    if \"Selenoprotein\" in new[i][3] or \"selenoprotein\" in new[i][3]:\n",
    "        print(new[i])\n",
    "        k = 0\n",
    "        while (k<maxbraker):\n",
    "            if k%3!=2:\n",
    "                if new[i][0] == braker[k][0][1:]:\n",
    "                    #print(new[i][0] + \" Sequence from Breaker:\\n\" + braker[k+1][0] + \"\\n\")\n",
    "                    numu = braker[k+1][0].count(\"C\")\n",
    "                    #print(\"C appeared: \" + str(numu) + \" times\\n\")\n",
    "            k+=1\n",
    "    i+=1\n",
    "manualresults = [\"g10478.t1 preserves the USS as CSS\", \n",
    "                 \"g10478.t2 preserves the USS as CSS\",\n",
    "                 \"g1679.t1 sequence is AFTER the selenocysteine is (see alignment)\",\n",
    "                 \"g2069.t1 is not very similar and the U is skipped but the rest of the protein is translated (see alignment)\",\n",
    "                 \"g5490.t1 cut off exactly where U is (mistranslation of selenocysteine likely)\",\n",
    "                 \"g5490.t2 cut off exactly where U is (mistranslation of selenocysteine likely)\",\n",
    "                 \"g6082.t1 cut off exactly where U is (mistranslation of selenocysteine likely)\",\n",
    "                 \"g7697.t1 preserves the first ASUYL as ASCGF, but the 9 selenocysteines in the C-terminus are truncated (see alignement)\",\n",
    "                 \"g9782.t1 sequence is AFTER the selenocysteine is (see alignment)\"\n",
    "                 ]\n",
    "for n in manualresults:\n",
    "    print(n)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "ac889a4e-53ed-42e0-b65b-4233930c2ec5",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['Xetrov107005635m', 'sp|Q8WWX9|SELM_HUMAN', 'SELENOM', 'Selenoprotein M ', '1.8e-40', '1.1e-40']\n",
      "['Xetrov107008048m', 'sp|P49908|SEPP1_HUMAN', 'SELENOP', 'Selenoprotein P ', '4.3e-58', '3e-58']\n",
      "['Xetrov107022849m', 'sp|Q9BVL4|SELO_HUMAN', 'SELENOO', 'Selenoprotein O ', '9.4e-284', '8e-283']\n",
      "['Xetrov107025172m', 'sp|Q9BQE4|SELS_HUMAN', 'SELENOS', 'Selenoprotein S ', '2.6e-57', '1.6e-57']\n",
      "['Xetrov107031781m', 'sp|P49908|SEPP1_HUMAN', 'SELENOP', 'Selenoprotein P ', '2.7e-55', '']\n",
      "['Xetrov107032545m', 'sp|O60613|SEP15_HUMAN', 'SELENOF', 'Selenoprotein F ', '4.8e-57', '1.1e-58']\n",
      "['Xetrov107034602m', 'sp|Q9Y6D0|SELK_HUMAN', 'SELENOK', 'Selenoprotein K ', '1e-38', '3.8e-39']\n",
      "['Xetrov107038964m', 'sp|P62341|SELT_HUMAN', 'SELENOT', 'Thioredoxin reductase-like selenoprotein T ', '8.5e-88', '3.8e-88']\n",
      "['Xetrov107041207m', 'sp|Q9NZV5|SELN_HUMAN', 'SELENON', 'Selenoprotein N ', '5.8e-185', '2.3e-186']\n",
      "['Xetrov107041767m', 'sp|Q9BVL4|SELO_HUMAN', 'SELENOO', 'Selenoprotein O ', '2.8e-101', '']\n",
      "Xetrov107005635m sequence skips over chunk containing U\n",
      "Xetrov107008048m sequence skips over all chunks containing the 10 Us\n",
      "Xetrov107022849m cut off exactly where U is (mistranslation of selenocysteine likely)\n",
      "Xetrov107025172m cut off exactly where U is (mistranslation of selenocysteine likely)\n",
      "Xetrov107031781m translates the first U correctly, but the second is missing and the sequence stops before the other 8 selenocysteines\n",
      "Xetrov107032545m sequence skips over chunk containing U\n",
      "Xetrov107034602m cut off very near where U is (mistranslation of selenocysteine likely)\n",
      "Xetrov107038964m sequence skips over chunk containing U\n",
      "Xetrov107041207m skips over chunk containing first U and cuts off exactly where the second U is (mistranslation of selenocysteine likely)\n",
      "Xetrov107041767m preserves the USS as CSS\n"
     ]
    }
   ],
   "source": [
    "# for 10.7\n",
    "i = 0\n",
    "while i < maxtenpseven:\n",
    "    if \"Selenoprotein\" in tenpseven[i][3] or \"selenoprotein\" in tenpseven[i][3]:\n",
    "        print(tenpseven[i])\n",
    "        k = 0\n",
    "        while (k<selenon):\n",
    "            selenogene = seleno[k][0][1:]\n",
    "            if tenpseven[i][0] == selenogene:\n",
    "                #print(seleno[k][0] + \" sequence from 10.7:\\n\" + seleno[k+1][0] + \"\\n\")\n",
    "                #numu = seleno[i+1][0].count(\"C\")\n",
    "                k = selenon\n",
    "            k+=1\n",
    "    i+=1\n",
    "manualresults = [\"Xetrov107005635m sequence skips over chunk containing U\",\n",
    "                 \"Xetrov107008048m sequence skips over all chunks containing the 10 Us\",\n",
    "                 \"Xetrov107022849m cut off exactly where U is (mistranslation of selenocysteine likely)\",\n",
    "                 \"Xetrov107025172m cut off exactly where U is (mistranslation of selenocysteine likely)\", \n",
    "                 \"Xetrov107031781m translates the first U correctly, but the second is missing and the sequence stops before the other 8 selenocysteines\",\n",
    "                 \"Xetrov107032545m sequence skips over chunk containing U\",\n",
    "                 \"Xetrov107034602m cut off very near where U is (mistranslation of selenocysteine likely)\",\n",
    "                 \"Xetrov107038964m sequence skips over chunk containing U\",\n",
    "                 \"Xetrov107041207m skips over chunk containing first U and cuts off exactly where the second U is (mistranslation of selenocysteine likely)\",\n",
    "                 \"Xetrov107041767m preserves the USS as CSS\"\n",
    "                 ]\n",
    "for n in manualresults:\n",
    "    print(n)\n",
    "    #pass"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "94fcde43-6ee2-442e-aa94-a9895dc553c9",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['Xetrov107003417m', 'sp|P36969|GPX4_HUMAN', 'GPX4', 'Phospholipid hydroperoxide glutathione peroxidase ', '7.7e-60', '1.5e-60']\n",
      ">Xetrov107003417m sequence from 10.7:\n",
      "MLNLLCRSIKRSVLLGSVIGGVQSRAMCAQVADWKAAKTIYEFSAVDIDGNEVSLEKYRGYVCIIVNEPGDEAQIKDFAA\n",
      "SYKVEFDMFSKIEVNGDGAHPLWKWMKEQPKGHGTLGNAIKWNFTKFLINREGAVVKRFSPMEDPVVIEKDLPNLL*\n",
      "\n",
      "['Xetrov107006413m', 'sp|Q9NNW7|TRXR2_HUMAN', 'TXNRD2', 'Thioredoxin reductase 2, mitochondrial ', '1.1e-256', '1.2e-256']\n",
      ">Xetrov107006413m sequence from 10.7:\n",
      "MKVGSVLRKAGLLGGIRHDYDLLVIGGGSGGLACAKQAAQFGKKVAVFDYVEPSPRGTKWGIGGTCVNVGCIPKKLMHQA\n",
      "ALIGSTLKDAPHYGWDTPNEIQHNWGKMAKAVQNYVKSLNWGHRIQLQDKKVKYFNMKASFVDEHCIRGVTKAGKETLVT\n",
      "AENIVIATGGRPKYPTHVPGALEYGITSDDLFWLKESPGKTLVVGASYVSLECAGFLTGIGLDTAAMVRSIPLRGFDQQM\n",
      "AYLVTDYMELHGTKFLWKCTPSNVEKLKNGKLEVTWKNTESGKEGADIYDTVMWAVGRAAETQYLNLEKVGVDVKSDTGK\n",
      "IIVNASEATSVPHIYAIGDITEGRPELTPTAIAAGKLLASRLFSGSAELMDYDSVPTTVFTPLEYGCVGISEEDAKERYG\n",
      "DDNIEVFHAFYKPLEFIVAERNGSQCYIKIICLRTQDQRILGLHFTGPNAGEVIQGFALGIKCGATYHQLMRTVGIHPTC\n",
      "AEEVTKLHITKRSGLDPTVTGC*\n",
      "\n",
      "['Xetrov107020855m', 'sp|P22352|GPX3_HUMAN', 'GPX3', 'Glutathione peroxidase 3 ', '5.4e-65', '4e-65']\n",
      ">Xetrov107020855m sequence from 10.7:\n",
      "MQYQELNALQEELKNNNFVILGFPSNQFGMQEPGRNDEILLGLEYVRPGGKFVPNFQLFEKGDINGRKEQKFYTFLKNSC\n",
      "PPVGDNFGSATNRLMWEPIKVNDVKWNFEKFLVGPDGRPVKRWLPRTPVAQVRREIMSYMKLQPGTQRLLMLGLEQK*\n",
      "\n",
      "['Xetrov107021964m', 'sp|Q16881|TRXR1_HUMAN', 'TXNRD1', 'Thioredoxin reductase 1, cytoplasmic ', '2e-303', '2e-303']\n",
      ">Xetrov107021964m sequence from 10.7:\n",
      "MGNSVKKGKAVFVDQTQVKSMRLYKAALEDQSSKSPEIPKCKTNENEDNGWRTDKNKSGLKIEIEEYICSSDVVIFSKTS\n",
      "CNSCAKVKDHFDQMGITYLTVELDQIENGHKLEELLYEQTMETSLPSVFVKEKHIGNYDDILKLSKENLFQKLLETKEDN\n",
      "PDRSGYDYDLIVIGGGSGGLAASKEAANYGKKVLVLDFVTPTPLGTKWGLGGTCVNVGCIPKKLMHQAALLRQALKDSQK\n",
      "YGWQIEDNIQHNWEIMTDSVLNYIGSLNFNYRVSLMQNNVKYENGYGEFVEPHTIKSTNSRGKEKYFTAEKFLIATGERP\n",
      "RYLGIPGDKEYCITSDDLFSLPYCPGKTLVVGASYVALECAGFLAGLGLDVTVMVRSILLRGFDQQMANMIGEYMEEHGV\n",
      "KFIKQCVPTKIEQIEAGTPGRLKVTSQASDGAETTEEYNTVLLAIGRDACTRNIGLEIPGVKINEKTGKIPVNDEEQTNV\n",
      "PYIYAIGDVLQDKLELTPVAIQAGRLLARRLYGDSTVKCDYVNVPTTVFTPLEYGACGLSEEHAIRQYEEENIEVYHSYF\n",
      "WPLEWTVPARDNNKCYAKIICNLKDKGRVVGFHVLSPNAGEITQGFAVAIKCGLTKDQLDNTIGIHPVCAEIFTTLTVTK\n",
      "RSGGNILQSGC*\n",
      "\n",
      "['Xetrov107032116m', 'sp|P49895|IOD1_HUMAN', 'DIO1', 'Type I iodothyronine deiodinase ', '4.9e-30', '4.4e-30']\n",
      ">Xetrov107032116m sequence from 10.7:\n",
      "MESLLQTIKLMVRFIQKTMIFFFLFIYVVVGKVLMFLFPQTMASVLKSRFETTGVHDPKFQYEDWGPTFFTYKFLRSVLE\n",
      "IMWLRLEDEAFVGHSAPNTPVIDLNGELHHIWDYLQGTRPLVLNFGSCT*\n",
      "\n",
      "['Xetrov107033615m', 'sp|P07203|GPX1_HUMAN', 'GPX1', 'Glutathione peroxidase 1 ', '2.1e-63', '1.9e-63']\n",
      ">Xetrov107033615m sequence from 10.7:\n",
      "MSRLQSMYGPRGLQVLAFPCNQFGHQENSGNQEILNILKHVRPGGGFEPNFPLFEKVDVNGEKEHPLFTFLKGQLPYPSD\n",
      "DSISLMQDPKSIIWSPVRRNDIAWNFEKFLIARNGVPYKRYGRRFETFNIQQDIEKLLDETCE*\n",
      "\n",
      "['Xetrov107034057m', 'sp|Q86VQ6|TRXR3_HUMAN', 'TXNRD3', 'Thioredoxin reductase 3 ', '0', '0']\n",
      ">Xetrov107034057m sequence from 10.7:\n",
      "MPPTGRELLQARVKELIDSNRVMVFSKSFCPYCDQVKELFSSLGAEYQSLELDECDDGAAIQETLHELTGQRTVPNVFVN\n",
      "KTHVGGCDKTLKAHKDGSLAKLLGDNSVTYDYDLIVIGGGSGGLACSKEAASFGKKVMVLDYVVPTPKGTSWGLGGTCVN\n",
      "VGCIPKKLMHQAAILGQSLKDSRKFGWEYEEQVKHNWDTMREAIQNYIGSLNWGYRVSLRDKQVRYENAYGEFVEPHKIK\n",
      "ATNKRGKESFFTAEKFVVATGERPRYLNIPGDKEYCITSDDLFSLPYCPGKTLVVGASYVALECAGFLAGLGLDATVMVR\n",
      "SILLRGFDQEMANRAGAYMETHGVKFIKQFVPISVELLEEGTPGRLKVTAKATEGDQIIEDEYNTVLIAVGRDACTRNIG\n",
      "LEKIGVKVNERNGKIPVSDEEQTSVPYVYAIGDILDGKLELTPVAIQAGRLLARRLYKGSKLKCDYVNVPTTVFTPLEYG\n",
      "CCGYAEEKAIEIYGEENLEVYHTLFWPLEWTVPSRDNNTCFAKIICNKQDDDRVVGFHVLGPNAGEITQGFGAAIKCGLT\n",
      "KEKLDETIGIHPTCAEVFTTMDVSKSSGGDITQKGC*\n",
      "\n",
      "['Xetrov107040419m', 'sp|Q9C0D9|EPT1_HUMAN', 'SELENOI', 'Ethanolaminephosphotransferase 1 ', '3.7e-187', '1.3e-186']\n",
      ">Xetrov107040419m sequence from 10.7:\n",
      "MHPFWNSIVRFFPTWLAPNLITFSGFLLLVFTFLLMAFFDPDFYASAPGQEHVPNWVWIVAGLLNFTAYTLDGVDGKQAR\n",
      "RTNSSTPLGELFDHGLDSWACIFFVVTVYSIFGRGETGVSAVVLYFLLWVVLFSFILSHWEKYNTGILFLPWGYDLSQVT\n",
      "ISCVYLVTAVVGAETWYQPVVFNILYRDLFTTMIVGCAVCVTLPMSLYNVFKGYRNNTLKHTSLYEAMLPIISPVLLFIL\n",
      "STLWILASPSKILEHHPRLFYFMVGTTFANITCQLIVCQMSNTRCRPLSWLLIPLTLAVVVSLSTVTPRDETLALVLLTV\n",
      "LVTLAHIHYGVNVVNQLSKHFNILPFSLRKPSTD*\n",
      "\n",
      "['Xetrov107044544m', 'sp|Q9C0D9|EPT1_HUMAN', 'SELENOI', 'Ethanolaminephosphotransferase 1 ', '3.7e-139', '']\n",
      ">Xetrov107044544m sequence from 10.7:\n",
      "MFNHRYITPDQLAGFDNYKYSAVDTNPLSVYVMQHLWNKVVKVVPLWIAPNLLTLSGFLIILVNYFLLCFFDWNYTASGV\n",
      "GHEHIPNWVWISAAVGNFLAYALDSIDGKHARRTQSSSPLGELFDHGLDSWAASVFPLSLFSIFGCDSGEAGLSTRTMYS\n",
      "VSCIVLFTFMLSHWEKYNTGVLFLPWAYDISQVSLTLVYILTAIVGVDAYHKPLFLNFYMTEILIALLTGCSLLIAFPQT\n",
      "LYNIYKAHQQKALLKTSLYEGLLPLISPTLLFTLLSVWVAFSPCGILSKQPRLVLWMVGVAFSNVTCRVIICQMSNTRSE\n",
      "LFHWLLLPLLLLVGASVSGLLGIMEEIAFLVFTLFVTAAHVHYGVCVGRQLSQHFNIYAFSLKKRGGN*\n",
      "\n",
      "['Xetrov107053184m', 'sp|Q99611|SPS2_HUMAN', 'SEPHS2', 'Selenide, water dikinase 2 ', '1.6e-194', '5.9e-195']\n",
      ">Xetrov107053184m sequence from 10.7:\n",
      "ELTAAPVLPKGIGLDSCVIPLRHRGLSLVQTTDFFYPSVEDPYMQGRIACANVLSDLYAMGITECDNMLMLLSVSQKMTE\n",
      "EEREKVTPLMIKGFRDAAEEGGTSVTGGQTVVNPWIIIGGVATVVCQANEFIMPQNAVAGDVLVLTKPLGTQVAVNAHQW\n",
      "LDNPERWNKIRLVVSREDVELAYQEAMFNMATLNRTAAALMHTFNAHAATDVTGFGILGHAQNLAQEQHNEVSFVIHNLP\n",
      "IIAKMAAITKACGNRFGLLQGTSPETSGGLLICLPREQAARFCAEIKSPKHGEGHQAWIIGIVEKGNRSARIIEKPRIIE\n",
      "VTPRGAASSDNSSMTSLDGPC*\n",
      "\n",
      "['Xetrov107055985m', 'sp|Q92813|IOD2_HUMAN', 'DIO2', 'Type II iodothyronine deiodinase ', '8.2e-117', '1.1e-118']\n",
      ">Xetrov107055985m sequence from 10.7:\n",
      "MGLLSVDLLITLQILPGFFSNCLFLALYDSVVLVKHVLLQLSRSKSSQSQWRRMLTPEGLRCVWNSFLLDAYKQVKLGQD\n",
      "APNSNVIQVSNKICKSVQRKLVGKCHLLDFASSERPLLPAFSKLVEEFSSVADFVLVYIDEAHPSDGWAAPGTTSYEVKK\n",
      "HRNQEERCAAASKLLEHFSIPPQCQVVADCMDNNANVAYGVSFERVCIVQKQKIVYLGGKGPFFYNIQEIRRWLELSFGK\n",
      "R*\n",
      "\n",
      "['Xetrov107056581m', 'sp|P55073|IOD3_HUMAN', 'DIO3', 'Thyroxine 5-deiodinase ', '3.4e-80', '9e-82']\n",
      ">Xetrov107056581m sequence from 10.7:\n",
      "MLHCAGPHTSKLLKQVAACCLLLPRFLLTALMLWLLDFQCIRRRVLVTAREESSPELEDPPLCVSDSNRMCTVESLRAVW\n",
      "HGQKLDYFKSAHLGSSAPNTEVVMLEGRRLCRILDFSQGKRPLEAHPSDGWVSTDASYQIPQHQCLQDRLAAAQLMLQGV\n",
      "PGCRVVADTMDNSSNAAYGAYFERLYIILEGKVVYQGGRGPEGYKISELRMWLEQYQQGTIGTSGTGHVVIQV*\n",
      "\n",
      "['Xetrov107057173m', 'sp|P18283|GPX2_HUMAN', 'GPX2', 'Glutathione peroxidase 2 ', '9.6e-57', '']\n",
      ">Xetrov107057173m sequence from 10.7:\n",
      "MTTVYIILKVNLKENCKNEEILNSLKYVRPGKGFVPGFTLFQKCDVNGKDTHSVFAYLKDKLPVPDNEPAALISDPRYIV\n",
      "WNPVHRSDISWNFEKFLIGPEGEPFKRYNKNFQTISIEPDIQRLLKLTK*\n",
      "\n",
      "['Xetrov107060523m', 'sp|Q9NZV6|MSRB1_HUMAN', 'MSRB1', 'Methionine-R-sulfoxide reductase B1 ', '2.5e-48', '1.9e-48']\n",
      ">Xetrov107060523m sequence from 10.7:\n",
      "MSFCSFFGGEVYKDHFKTGIYVCSECNYELFSSRSKYQHSSPWPAFTETVHKDSISKYLERPNAYKVSCGKCGNGLGHEF\n",
      "INDGPKKGQSRF*\n",
      "\n",
      "\n",
      "Human proteins that don't have Xenopus orthologs:\n",
      "sp|P59796|GPX6_HUMAN Glutathione peroxidase 6 \n",
      "sp|P59797|SELV_HUMAN Selenoprotein V \n",
      "sp|P63302|SELW_HUMAN Selenoprotein W \n",
      "sp|Q8IZQ5|SELH_HUMAN Selenoprotein H \n"
     ]
    }
   ],
   "source": [
    "# turns out there may be more selenocysteine-containing proteins than expected!\n",
    "i = 0\n",
    "j = 0\n",
    "k = 0\n",
    "mappedlist = []\n",
    "while i < maxtenpseven:\n",
    "    k = 0\n",
    "    while k < maxhumanseleno:\n",
    "        if k%3==0:\n",
    "            human_entry = humanseleno[k][0].split(\" \")[0][1:]\n",
    "            if human_entry == tenpseven[i][1]:\n",
    "                mappedlist.append(human_entry)\n",
    "            # only processing ones that were missed in earlier sections\n",
    "                if \"Selenoprotein\" not in tenpseven[i][3] and \"selenoprotein\" not in tenpseven[i][3]:\n",
    "                    print(tenpseven[i])\n",
    "                    j = 0\n",
    "                    while j < maxxentraa:\n",
    "                        if xentraa[j][0][0] == \">\":\n",
    "                            xetrov = xentraa[j][0].split(\" \")[0][1:]\n",
    "                            if tenpseven[i][0] == xetrov:\n",
    "                                #print(\">\" + human_entry + \"\\n\")\n",
    "                                print(\">\" + xetrov + \" sequence from 10.7:\")\n",
    "                                x = 1\n",
    "                                #need to change to print ALL lines\n",
    "                                while xentraa[j+x][0][0] != \">\":\n",
    "                                    print(xentraa[j+x][0])\n",
    "                                    x+=1\n",
    "                                print(\"\")\n",
    "                        j+=1       \n",
    "                    k = maxhumanseleno\n",
    "        k+=1\n",
    "    i+=1\n",
    "\n",
    "k = 0\n",
    "print(\"\\nHuman proteins that don't have Xenopus orthologs:\")\n",
    "while k < maxhumanseleno:\n",
    "    if k%3==0:\n",
    "        human_entry = humanseleno[k][0].split(\" \")[0][1:] \n",
    "        if human_entry not in mappedlist:\n",
    "            print(humanseleno[k][0].split(\"OS=\")[0][1:])\n",
    "    k+=1\n",
    "\n",
    "manualres = [\"Xetrov107003417m sequence skips over chunk containing U\",\n",
    "             \"Xetrov107006413m cuts off exactly where U is (mistranslation of selenocysteine likely)\",\n",
    "             \"Xetrov107020855m sequence skips over chunk containing U\",\n",
    "             \"Xetrov107021964m cuts off exactly where U is (mistranslation of selenocysteine likely)\",\n",
    "             \"Xetrov107032116m cuts off exactly where U is (mistranslation of selenocysteine likely)\",\n",
    "             \"Xetrov107033615m sequence skips over chunk containing U\",\n",
    "             \"Xetrov107034057m cuts off exactly where U is (mistranslation of selenocysteine likely)\",\n",
    "             \"Xetrov107040419m cuts off exactly where U is (mistranslation of selenocysteine likely)\",\n",
    "             \"Xetrov107044544m cuts off exactly where U is (mistranslation of selenocysteine likely)\",\n",
    "             \"Xetrov107053184m sequence skips over chunk containing U\",\n",
    "             \"Xetrov107055985m skips over the first U and stops exactly where the second U is (mistranslation of selenocysteine likely)\",\n",
    "             \"Xetrov107056581m sequence skips over chunk containing U\",\n",
    "             \"Xetrov107057173m sequence skips over chunk containing U\",\n",
    "             \"Xetrov107060523m cuts off exactly where U is (mistranslation of selenocysteine likely)\"\n",
    "            ]      "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "bdb7a2b5-849c-4298-8de0-fd1738ebd63a",
=======
   "cell_type": "code",
   "execution_count": null,
   "id": "7fa182cd-9749-497a-9caf-247be5e23de3",
>>>>>>> 85b12dc514aed880607e74140170a6fe33e6f95d
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.13"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
